src/provider/view/components/NotificationCard/NotificationCard.tsx

```diff
@@ -1,67 +1,106 @@
 import * as React from 'react';
 
 import {NotificationTime} from '../NotificationTime/NotificationTime';
-import {Button} from '../Button/Button';
+import {Button} from '../Controls/Button/Button';
 import {StoredNotification} from '../../../model/StoredNotification';
-import {CloseButton} from '../CloseButton/CloseButton';
+import {CircleButton, IconType} from '../CircleButton/CircleButton';
+import {Actionable, MinimizeToast} from '../../../store/Actions';
 import {RemoveNotifications, ClickButton, ClickNotification} from '../../../store/Actions';
-import {Actionable} from '../../containers/NotificationCenterApp';
+import {useOnClickOnly} from '../../hooks/clicks';
 
-interface NotificationCardProps extends Actionable {
+import {Body} from './Body';
+import {Loading} from './Loading';
+
+import './NotificationCard.scss';
+
+interface Props extends Actionable {
     notification: StoredNotification;
+    isToast?: boolean;
 }
 
-export function NotificationCard(props: NotificationCardProps) {
-    const {notification, storeDispatch} = props;
+NotificationCard.defaultProps = {
+    isToast: false
+};
+
+export function NotificationCard(props: Props) {
+    const {notification, storeDispatch, isToast} = props;
     const data = notification.notification;
+    const [loading, setLoading] = React.useState(false);
+    const cardRef = React.createRef<HTMLDivElement>();
+    const [validClick, finishedClick] = useOnClickOnly(cardRef);
 
     const handleNotificationClose = () => {
         storeDispatch(new RemoveNotifications([notification]));
     };
 
-    const handleButtonClick = (buttonIndex: number) => {
+    const handleNotificationDismiss = () => {
+        if (isToast) {
+            storeDispatch(new MinimizeToast(notification));
+        }
+    };
+
+    const handleButtonClick = async (buttonIndex: number) => {
         // TODO: Have RemoveNotifications dispatched from inside ClickButton [SERVICE-623]
-        storeDispatch(new ClickButton(notification, buttonIndex));
-        storeDispatch(new RemoveNotifications([notification]));
+        await storeDispatch(new ClickButton(notification, buttonIndex));
+        await storeDispatch(new RemoveNotifications([notification]));
```

Comment:
I've added some CSS to make notifications take no events after it has been interacted with to counter this problem. And I'll do some hardening in #127 to prevent double DB writes.

Reply:
Think do this as Promise.all([...]), so the click and remove from the store happen synchronously

Reply:
Done